### PR TITLE
API doc for core directives

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

Add API doc for directives in a single file, core/src/metadata/directives.ts.

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:
 

Issue Number:  #24896

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
 Existing PR #24135 contains the fix requested by the issue, but is too big to review and merge easily.
This copies the affected file to a PR of more limited in scope that should be easier to land.

### CODE REVIEW

**Comment by jenniferfell** [comment]
LGTM

This PR:
* Moves the existing content about "preserving whitespace" so that it will appear in the AIO API doc for "Component" using the new templates. This fixes a problem introduced by the new API templates i v6, which inadvertently hid this content. 
* Removed usage notes about animations. This content is more appropriately and completely covered in the Animations guide: https://pr26923-486b6b1.ngbuilds.io/guide/animations#triggering-the-animation

**Review by jenniferfell — APPROVED** [review]
LGTM

**Comment by jenniferfell** [comment]
@petebacondarwin : Because this is technically a code change, a second pair of eyes would be great. Thanks!

File: packages/core/src/metadata/directives.ts [hunk 1]
```diff
@@ -436,6 +436,63 @@ export interface ComponentDecorator {
...
+   *
+   * Removing whitespace can greatly reduce AOT-generated code size and speed up view creation.
+   * As of Angular 6, the default for `preserveWhitespaces` is false (whitespace is removed).
+   * To change the default setting for all components in your application, set
+   * the `preserveWhitespaces` option of the AOT compiler.
+   *
+   * By default, the AOT compiler removes whitespace characters as follows:
+   * * Trims all whitespaces at the beginning and the end of a template.
+   * * Removes whitespace-only text nodes. For example,
+   * ```
```
Comment (petebacondarwin) [inline packages/core/src/metadata/directives.ts]: There should be blank lines around backtick blocks

**Review by petebacondarwin — APPROVED** [review]

**Comment by jbogarthyde** [comment]
Caretaker -- approved and passing tests. Can you run G3 if needed?

**Pull request closed without merging**